homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,185 @@
+            title=user_input.get(CONF_NAME, CONF_HOST), data=data
+        )
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered Webostv device."""
+        user_input = {
+            CONF_HOST: urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname,
+            CONF_NAME: discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME],
+        }
+        self.context["title_placeholders"] = {"name": user_input[CONF_NAME]}
+        self._force_pairing = True
+        return await self.async_step_user(user_input)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.options = config_entry.options
+        self.host = config_entry.data[CONF_HOST]
+        self.key = config_entry.data[CONF_CLIENT_SECRET]
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+        if user_input is not None:
+            on_action = user_input.get(CONF_ON_ACTION)
+
+            if on_action:
+                action_state = self.hass.states.get(on_action)
+
+                if not action_state or action_state.domain != "script":
+                    errors["base"] = "script_not_found"
+
+            if "base" not in errors:
+                options_input = {
+                    CONF_ON_ACTION: on_action,
+                    CONF_SOURCES: user_input[CONF_SOURCES],
+                }
+                return self.async_create_entry(title="", data=options_input)
+
+        # Get turn on service
+        script_turn_on = self.options.get(CONF_ON_ACTION, "")
+
+        # Get sources
+        sources = self.options.get(CONF_SOURCES, "")
+        sources_list = await async_default_sources(self.host, self.key)
+        if sources_list is None:
+            errors["base"] = "cannot_retrieve"
+
+        options_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_ON_ACTION,
+                    description={"suggested_value": script_turn_on},
+                ): str,
+                vol.Optional(
+                    CONF_SOURCES,
+                    description={"suggested_value": sources},
+                ): cv.multi_select(sources_list),
+            }
+        )
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=options_schema,
+            errors=errors,
+            description_placeholders={"docs_url": ON_ACTION_DOCS},
+        )
+
+
+async def async_default_sources(host, key) -> list:
+    """Construct sources list."""
+    sources = []
+    try:
+        client = await async_control_connect(host, key)
+    except CannotConnect:
+        return None
+
+    for app in client.apps.values():
+        sources.append(app["title"])
+    for source in client.inputs.values():
+        sources.append(source["label"])
```

Comment:
```suggestion
    sources = [
        *(app["title"] for app in client.apps.values()),
        *(app["label"] for app in client.inputs.values())
    ]
```

Reply:
untested, but something like this should work

Reply:
Tested, working correctly